app/models/like.rb

```diff
@@ -1,4 +1,11 @@
 class Like < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  after_save :update_like_counter
+  def update_like_counter
+    post = Post.find_by_id(post_id)
+    post.likes_counter = post.likes_counter.to_i + 1
+    post.save
+  end
```

Comment:
- [ ] Kindly use `increment!` instead of manually incrementing the counter. Rails provide a convenient method called `increment!` that can increment a counter attribute and save the record in a single operation. This simplifies the code and ensures atomicity. By using `increment!`, you avoid the need to load the entire User record, make changes, and save it again.

```suggestion
def update_post_counter
  author.increment!(:posts_counter)
end

```